2.4/run-mongod.sh

```diff
@@ -14,108 +15,95 @@ export MONGODB_DATADIR=/var/lib/mongodb/data
 export MONGODB_NOPREALLOC=${MONGODB_NOPREALLOC:-true}
 export MONGODB_SMALLFILES=${MONGODB_SMALLFILES:-true}
 export MONGODB_QUIET=${MONGODB_QUIET:-true}
-
 export MONGODB_CONFIG_PATH=/var/lib/mongodb/mongodb.conf
-export MONGODB_PID_FILE=var/run/mongodb/mongodb.pid
-
-MAX_ATTEMPTS=60
-SLEEP_TIME=1
+export MONGODB_PID_FILE=/var/lib/mongodb/mongodb.pid
 
 function usage() {
-	echo "You must specify following environment variables:"
-	echo "  \$MONGODB_USERNAME"
-	echo "  \$MONGODB_PASSWORD"
-	echo "  \$MONGODB_DATABASE"
-	echo "Optional variables:"
-	echo "  \$MONGODB_ADMIN_PASSWORD"
-	echo "MongoDB settings:"
-	echo "  \$MONGODB_NOPREALLOC (default: true)"
-	echo "  \$MONGODB_SMALLFILES (default: true)"
-	echo "  \$MONGODB_QUIET (default: false)"
-	exit 1
-}
-
-function up_test() {
-	for i in $(seq $MAX_ATTEMPTS); do
-		echo "=> Waiting for confirmation of MongoDB service startup"
-		set +e
-		mongo admin --eval "help"
-		status=$?
-		set -e
-		if [ $status -eq 0 ]; then
-			echo "=> MongoDB service has started"
-			return 0
-		fi
-		sleep $SLEEP_TIME
-	done
-	echo "=> Giving up: Failed to start MongoDB service"
-	exit 1
-	}
-
-function down_test() {
-	for i in $(seq $MAX_ATTEMPTS); do
-		echo "=> Waiting till MongoDB service is stopped"
-		set +e
-		mongo admin --eval "help"
-		status=$?
-		set -e
-		if [ $status -ne 0 ]; then
-			echo "=> MongoDB service has stopped"
-			return 0
-		fi
-		sleep $SLEEP_TIME
-	done
-	echo "=> Giving up: Failed to stop MongoDB service"
-	exit 1
+  echo "You must specify following environment variables:"
+  echo "  MONGODB_USERNAME"
+  echo "  MONGODB_PASSWORD"
+  echo "  MONGODB_DATABASE"
+  echo "Optional variables:"
+  echo "  MONGODB_ADMIN_PASSWORD"
+  echo "  MONGODB_REPLICA_NAME (if set, all required variables are ignored)"
+  echo "MongoDB settings:"
+  echo "  MONGODB_NOPREALLOC (default: true)"
+  echo "  MONGODB_SMALLFILES (default: true)"
+  echo "  MONGODB_QUIET (default: false)"
+  exit 1
 }
 
 # Make sure env variables don't propagate to mongod process.
 function unset_env_vars() {
-	unset MONGODB_USERNAME MONGODB_PASSWORD MONGODB_DATABASE MONGODB_ADMIN_PASSWORD
+  unset MONGODB_USERNAME MONGODB_PASSWORD MONGODB_DATABASE MONGODB_ADMIN_PASSWORD
 }
 
 function create_mongodb_users() {
-	# Start MongoDB service with disabled database authentication.
-	mongod -f $MONGODB_CONFIG_PATH &
+  # Start MongoDB service with disabled database authentication.
+  mongod -f $MONGODB_CONFIG_PATH --noprealloc --smallfiles --oplogSize 64 &
```

Comment:
Whats the point it adding `--noprealloc --smallfiles` since they are defined in config file ?